### ISSUE TITLE
False negative reportPossiblyUnboundVariable after pattern matching and enclosing scope

### ISSUE THREAD

**Issue by Etienne-Leveque** [microsoft/pyright#7384]
**Describe the bug**
Unbound variable is not detected in a local function that contains a condition with a logic operator, but detected if no logic operator.

**Code or Screenshots**
```python
from typing import Literal

def foo(lit: Literal["bar", "baz"]) -> None:
    v: int | None
    match lit:
        case "bar":
            v = 1
        case "baz":
            pass

    def with_logic_operator() -> bool:
        return v is None or v == 1  # no error

    def without_logic_operator() -> bool:
        return v is None  # "v" is possibly unbound

    _ = with_logic_operator()
    _ = without_logic_operator()
```

`foo("baz")` fails with "NameError: cannot access free variable 'v' where it is not associated with a value in enclosing scope", but no static error is detected.

**VS Code extension or command-line**
pyright CLI 1.1.352

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll investigate further.

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.
